lib/python-indent.coffee

```diff
@@ -75,32 +78,58 @@ module.exports = PythonIndent =
 
     # I'm glad Atom has an optional `column` param to subtract spaces from
     # soft-tabs, though I don't see it used anywhere in the core.
+    # It looks like for hard tabs, the "tabs" input can be fractional and
+    # the "column" input is ignored...?
     indent = editor.buildIndentString(tabs, column=offset)
 
-    # Set the indent
-    editor.getBuffer().setTextInRange([[row, 0], [row, 0]], indent)
-
-  unindentOnOpeningDelimiter: (editor, row, previousLine) ->
-    # Get all preceding lines
-    lines = if row > 0 then editor.buffer.lines[0..row - 1] else []
-
-    # Loop in reverse through lines
-    for line, i in lines by -1
-
-      # Set indent for row after declaration of block
-      if PythonIndent.openingDelimiterIndentRegex.test line
-
-        # Indent one tab-level past declaration
-        indent = editor.indentationForBufferRow i
-        indent += 1 if previousLine.slice(-1) is ':'
-        editor.setIndentationForBufferRow row, indent
-
-        # Stop trying after success
-        break
-
-  indentHanging: (editor, row, previousLine) ->
-    # Indent at the current block level plus the setting amount (1 or 2)
-    indent = (editor.indentationForBufferRow row) + (atom.config.get 'python-indent.hangingIndentTabs')
-
-    # Set the indent
-    editor.setIndentationForBufferRow row, indent
+    # The range of text to replace with our indent
+    # will need to change this for hard tabs, especially tricky for when
+    # hard tabs have mixture of tabs + spaces, which they can judging from
+    # the editor.buildIndentString function
```

Comment:
Yes, that function allows for a mixture of hard tabs and spaces, but PEP8 does not, if I remember correctly. The interpreter might even fail when tabs and spaces are mixed.